### PR TITLE
Settings subsections refined w/ translation addition

### PR DESCRIPTION
**Pull request by benalleng**
Using #524 as a reference I tuned the settings page to be better organized. Additionally I did a small amount of translation for the settings page as well added some unfinished translations in the french translation.json to make it easier to do translations moving forward where it doesn't yet exist. Finally I added little country flags to the languages.js file.  While there are only two languages now I think moving forward having a quick way for people to see the flag reference as the list of languages gets longer it could be helpful to use identifying flags.

image

image

### CODE REVIEW

**Comment by benalleng** [comment]
Feedback is welcome! I'll start.. Even though the user is led there by a classic cog symbol, I wonder if it would be worth keeping a main "Settings" title at the very top before the first section

**Comment by theborakompanioni** [comment]
Hey @benalleng!

Really nice. Thank you!

Two things from my side:
- If you are okay with it, let's remove the flags from the language dropdown. e.g.  French is spoken in many countries, not exclusively in France. Same with English and many other languages. I'd vote for removing them. Just let me know what you think.
- Some translations in `fr/translation.json`  are in English. It's easier to see the progress on Transifex, if this file only contains actually translated keys, e.g. French has 68% translated keys. Missing keys will fallback to English anyway!


> Feedback is welcome! I'll start.. Even though the user is led there by a classic cog symbol, I wonder if it would be worth keeping a main "Settings" title at the very top before the first section

My 2 sats: I think both is fine. We can keep it this way for now and if you are not happy with it, add a main header in a follow-up PR, if you so please!


What do you think? Let me know if you want to make the adaptions yourself.
Once done, I think it is ready to be merged. :rocket:

**Comment by benalleng** [comment]
- Ok! sounds good to me, as I look at it more I agree that as long as we continue to use the proper script for each language, flags aren't really needed or preferred.
- sorry about the Transifex thing, I didn't realize that was the proper method to go through translations. For the future I'll make sure to help with translations there :sweat_smile:  I'll also remove the other translations that already fallback to english!

File: src/components/Settings.test.jsx [hunk 1]
```diff
@@ -37,8 +37,11 @@ describe('<Settings />', () => {
     expect(screen.queryByText(/settings.button_switch_wallet/)).toBeVisible()
 
     expect(screen.getByText('settings.section_title_community')).toBeVisible()
-    expect(screen.queryByText(/settings.github/)).toBeVisible()
     expect(screen.queryByText(/settings.telegram/)).toBeVisible()
     expect(screen.queryByText(/settings.jm_twitter/)).toBeVisible()
+
+    expect(screen.getByText('settings.section_title_community')).toBeVisible()
+    expect(screnn.queryByText(/settings.documentation/)).toBeVisible()
```
Comment (theborakompanioni) [inline src/components/Settings.test.jsx:44]: Small typo.

```suggestion
    expect(screen.queryByText(/settings.documentation/)).toBeVisible()
```

File: src/i18n/locales/fr/translation.json [hunk 2]
```diff
@@ -144,9 +148,34 @@
...
+      "text_button_submit": "Sauvegarder",
+      "text_button_submitting": "Sauvant...",
+      "error_loading_fee_config_failed": "Error while loading fee config values. ",
+      "error_saving_fee_config_failed": "Error while saving fee config values. Reason: {{ reason }}"
+    }
+  },
+  "logs": {
+    "title": "Logs",
+    "error_loading_logs_failed": "Could not retrieve log file."
   },
```
Comment (theborakompanioni) [inline src/i18n/locales/fr/translation.json:179]: Removing some untranslated keys:

```suggestion
    "documentation": "Documentation",
    "github": "GitHub",
    "matrix": "Matrix",
    "telegram": "Telegram",
    "jm_twitter": "JoinMarket Twitter",
    "jam_twitter": "Jam Twitter",
    "confirm_locking_modal_title": "Verrouiller le Portefeuille",
    "fees": {
      "text_button_cancel": "Annuler",
      "text_button_submit": "Sauvegarder",
      "text_button_submitting": "Sauvant..."
    }
  },
```

**Comment by theborakompanioni** [comment]
> * Ok! sounds good to me, as I look at it more I agree that as long as we continue to use the proper script for each language, flags aren't really needed or preferred.

:+1:

> 
>     * sorry about the Transifex thing, I didn't realize that was the proper method to go through translations. For the future I'll make sure to help with translations there sweat_smile  I'll also remove the other translations that already fallback to english!

Don't worry!

Two small suggestions (one typo keeping the tests from succeeding, and removing left-over untranslated keys from `fr`), then we can push it over the finish line. :muscle:

**Review by theborakompanioni — APPROVED** [review]
I just now saw, that I can apply the suggestions myself via the UI.
Thank you @benalleng for your work :pray: 

Looking forward to your next contribution :orange_heart: 

Closes #524.

**Comment by benalleng** [comment]
Awesome! exciting to have my code merged! A question for the future, how do I conduct tests before I submit a pull request so that i can be certain you have less work to complete the merges on your end. Still a pretty new coder and I'm not too familiar with running tests :sweat_smile: Thanks for your help in getting it all finished, excited to sink my teeth into more issues moving forward!

**Comment by theborakompanioni** [comment]
> Awesome! exciting to have my code merged! 

:muscle: 

Thanks again for your work. Every little improvement is important in open source development. Your contribution is invaluable for projects like Jam.

> A question for the future, how do I conduct tests before I submit a pull request so that i can be certain you have less work to complete the merges on your end. Still a pretty new coder and I'm not too familiar with running tests sweat_smile Thanks for your help in getting it all finished, excited to sink my teeth into more issues moving forward!

Don't worry too much. We can figure it out during review and no code is perfect at first try. You can try running `npm test` before submitting your work - it's cool that you already adapted the tests in this PR in the first place ;-)

To run all tests (non-interactively) do `CI=true npm test`.

**Comment by dergigi** [comment]
Fantastic! Thank you for your contribution @benalleng 🙏🧡

**Pull request merged**